pyccel/codegen/build_generation/build_gen.py

```diff
@@ -0,0 +1,37 @@
+"""
+Module describing things that are in common between different build system handlers.
+"""
+
+class BuildSystemHandler:
+    """
+    The superclass from which build system handlers should inherit.
+
+    The superclass from which build system handlers should inherit.
+    This class exists to ensure that all build system handlers have
+    the same constructor.
+
+    Parameters
+    ----------
+    pyccel_dir : Path
+        The directory where generated files should be outputted.
+    root_dir : Path
+        The directory from which the pyccel-make command was called.
+    output_dir : Path
+        The directory where the final files should be outputted.
+    verbose : int
+        The level of verbosity.
+    debug_mode : bool
+        Indicates if we should compile in debug mode.
+    compiler : Compiler
```

Comment:
I suggest mentioning that the `Compiler` class is defined in module `pyccel.codegen.compilers.compiling`

Reply:
Fixed [0ba718a](https://github.com/pyccel/pyccel/pull/2383/commits/0ba718af1f0a25f43c90008642ca376078419bfd)